### ISSUE TITLE
Very slow sync

### ISSUE THREAD

**Issue by rex4539** [ZcashFoundation/zebra#1516]
Steps:
Start sync from scratch.

What happened:
At some point, sync started to go painfully slow.

[debug.log](https://github.com/ZcashFoundation/zebra/files/5687141/debug.log)

Expected result:
Sync is fast.

**Comment by teor2345** [comment]
Duplicate of #1435

The logs show the same symptoms: many peer connections have failed, so Zebra only has 2 peers left. If you leave it running long enough, it should permanently fail.
```
Dec 14 09:21:51.916  INFO {zebrad="e537926e"}:sync:obtain_tips: zebra_network::peer_set::set: network request with no ready peers: finding more peers, waiting for 2 peers to answer requests
```

**Comment by teor2345** [comment]
(There isn't much extra info here, and I can't seem to get GitHub's duplicate issues to work, so I'm just going to close this issue.)

**Comment by teor2345** [comment]
Please re-test with the latest Zebra `main` branch, we've merged the #1468 peer diagnostics and #1511 panic fix since then.
